backend/ccm/canvas_api/canvas_credential_manager.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from http import HTTPStatus
+from django.conf import settings
+from canvas_oauth.oauth import get_oauth_token, handle_missing_token
+from rest_framework.request import Request
+from canvas_oauth.models import CanvasOAuth2Token
+from canvas_oauth.exceptions import InvalidOAuthReturnError
+
+from canvasapi import Canvas
+from canvasapi.exceptions import (
+    BadRequest, CanvasException, Conflict, Forbidden, InvalidAccessToken, RateLimitExceeded,
+    ResourceDoesNotExist, Unauthorized, UnprocessableEntity
+)
+from .exceptions import CanvasHTTPError 
+
+logger = logging.getLogger(__name__)
+
+class CanvasCredentialManager:
+
+  EXCEPTION_STATUS_MAP = {
```

Comment:
I think this would be more organized for this and `canvas_api_exception` method to be in `exceptions.py`.

I'm kind of even confused about what this does. Maybe you can describe why we need this custom MAP and exception handling or add some comments around this?

Reply:
The canvasapi return only success/error object, in CCM response to API calls from FE we return a status code and based on the status code different action are taken in FE. 
So we are keeping the internal map, I don't think the response Object from canvasapi return status code. Hence the map

Reply:
I added comment on this.